src/index.js

```diff
@@ -8,30 +8,51 @@ import {runScripts} from './transformScriptTags';
 function processOptions(options) {
   // Parse preset names
   const presets = (options.presets || []).map(presetName => {
-    if (typeof presetName === 'string') {
-      var preset = availablePresets[presetName];
-      if (!preset) {
-        throw new Error(`Invalid preset specified in Babel options: "${presetName}"`);
+    let preset = null;
+
+    if (presetName instanceof Array && typeof presetName[0] === 'string') {
+      if (presetName[0] in availablePresets) {
+        preset = [availablePresets[presetName[0]]].concat(presetName.slice(1));
+
+        // workaround for babel issue
+        // at some point, babel copies the preset, losing the non-enumerable
+        // buildPreset key; convert it into an enumerable key.
+        if (typeof preset[0] === 'object' && 'buildPreset' in preset[0]) {
+          preset[0] = { ...preset[0], buildPreset: preset[0].buildPreset }
+        }
       }
-      return preset;
+    } else if (typeof presetName === 'string') {
+      preset = availablePresets[presetName];
     } else {
       // Could be an actual preset module
       return presetName;
     }
+
+    if (!preset) {
+      throw new Error(`Invalid preset specified in Babel options: "${presetName}"`);
+    }
+    return preset;
   });
 
   // Parse plugin names
   const plugins = (options.plugins || []).map(pluginName => {
-    if (typeof pluginName === 'string') {
-      var plugin = availablePlugins[pluginName];
-      if (!plugin) {
-        throw new Error(`Invalid plugin specified in Babel options: "${pluginName}"`);
+    let plugin = null;
+
+    if (pluginName instanceof Array && typeof pluginName[0] === 'string') {
+      if (pluginName[0] in availablePlugins) {
+        plugin = [availablePlugins[pluginName[0]]].concat(pluginName.slice(1));
       }
-      return plugin;
+    } else if (typeof pluginName === 'string') {
+      plugin = availablePlugins[pluginName];
     } else {
       // Could be an actual plugin module
       return pluginName;
     }
```

Comment:
A lot of the logic in here is exactly the same as the logic in the preset parsing code. Could you please instead make a function that both of them use? Should be able to do something like `processPluginsOrPresets(pluginName, availablePlugins)`